watertap/examples/flowsheets/case_studies/BSM2/ASM_ADM_integration.py

```diff
@@ -0,0 +1,422 @@
+        has_phase_equilibrium=False,
+        outlet_state_defined=True,
+    )
+
+    m.fs.CL = Separator(
+        property_package=m.fs.props_ASM1,
+        outlet_list=["underflow", "effluent"],
+        split_basis=SplittingType.componentFlow,
+    )
+
+    m.fs.TU = Thickener(property_package=m.fs.props_ASM1)
+    m.fs.DU = DewateringUnit(property_package=m.fs.props_ASM1)
+
+    m.fs.MX2 = Mixer(
+        property_package=m.fs.props_ASM1, inlet_list=["feed_water1", "recycle1"]
+    )
+    m.fs.MX3 = Mixer(
+        property_package=m.fs.props_ASM1, inlet_list=["feed_water2", "recycle2"]
+    )
+    m.fs.MX4 = Mixer(
+        property_package=m.fs.props_ASM1, inlet_list=["thickener", "clarifier"]
+    )
+
+    # Clarifier
+    # TODO: Update once more detailed model available
+    m.fs.CL.split_fraction[0, "effluent", "H2O"].fix(0.993)
+    m.fs.CL.split_fraction[0, "effluent", "S_I"].fix(0.993)
+    m.fs.CL.split_fraction[0, "effluent", "S_S"].fix(0.993)
+    m.fs.CL.split_fraction[0, "effluent", "X_I"].fix(0.5192)
+    m.fs.CL.split_fraction[0, "effluent", "X_S"].fix(0.5192)
+    m.fs.CL.split_fraction[0, "effluent", "X_BH"].fix(0.5192)
+    m.fs.CL.split_fraction[0, "effluent", "X_BA"].fix(0.5192)
+    m.fs.CL.split_fraction[0, "effluent", "X_P"].fix(0.5192)
+    m.fs.CL.split_fraction[0, "effluent", "S_O"].fix(0.993)
+    m.fs.CL.split_fraction[0, "effluent", "S_NO"].fix(0.993)
+    m.fs.CL.split_fraction[0, "effluent", "S_NH"].fix(0.993)
+    m.fs.CL.split_fraction[0, "effluent", "S_ND"].fix(0.993)
+    m.fs.CL.split_fraction[0, "effluent", "X_ND"].fix(0.5192)
+    m.fs.CL.split_fraction[0, "effluent", "S_ALK"].fix(0.993)
+    m.fs.RADM.volume_liquid.fix(3400)
+    m.fs.RADM.volume_vapor.fix(300)
+    m.fs.RADM.liquid_outlet.temperature.fix(308.15)
+
+    # Apply scaling
+    m.fs.stream2adm = Arc(
+        source=m.fs.RADM.liquid_outlet, destination=m.fs.adm_asm.inlet
+    )
+    m.fs.stream6adm = Arc(source=m.fs.SP6.waste, destination=m.fs.TU.inlet)
+    m.fs.stream3adm = Arc(source=m.fs.TU.underflow, destination=m.fs.MX4.thickener)
+    m.fs.stream7adm = Arc(source=m.fs.TU.overflow, destination=m.fs.MX3.recycle2)
+    m.fs.stream9adm = Arc(source=m.fs.CL.underflow, destination=m.fs.MX4.clarifier)
+    m.fs.stream4adm = Arc(source=m.fs.adm_asm.outlet, destination=m.fs.DU.inlet)
+    m.fs.stream5adm = Arc(source=m.fs.DU.overflow, destination=m.fs.MX2.recycle1)
+    m.fs.stream01 = Arc(source=m.fs.FeedWater.outlet, destination=m.fs.MX2.feed_water1)
+    m.fs.stream02 = Arc(source=m.fs.MX2.outlet, destination=m.fs.MX3.feed_water2)
+    m.fs.stream03 = Arc(source=m.fs.MX3.outlet, destination=m.fs.CL.inlet)
+    m.fs.stream04 = Arc(source=m.fs.CL.effluent, destination=m.fs.MX1.feed_water)
+    m.fs.stream10adm = Arc(source=m.fs.MX4.outlet, destination=m.fs.asm_adm.inlet)
+    m.fs.stream1adm = Arc(source=m.fs.asm_adm.outlet, destination=m.fs.RADM.inlet)
+    pyo.TransformationFactory("network.expand_arcs").apply_to(m)
```

Comment:
In terms of organization, I would break this down into the following functions

1. build_flowsheet: build all components, add Vars/constraints desired, define/transform Arcs
2. set_operating_conditions: fix all variables
3. initialize_system: initialize the flowsheet

Then you can have a main function that runs all of these; also would need to add `calculate_scaling_factors`

Then once we add costing, we'd make another function, `add_costing`, where all of that'd be added to the flowsheet. Sometimes the flowsheet solves without costing added and could sometimes fail if flowsheet includes costing upfront due to some oversights. Breaking things up will make for streamlined troubleshooting later and may also be good for streamlining the jupyter notebook